app/src/main/AndroidManifest.xml

```diff
@@ -13,25 +13,24 @@
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
         android:theme="@style/AppTheme">
-        <activity android:name=".ViewAnotherUserActivity" />
-        <activity android:name=".InvitationConfirmationActivity" />
-        <activity android:name=".SupervisedGroupPageActivity" />
-        <activity android:name=".SupervisedGroupListActivity" />
-        <activity android:name=".MonitoringGroupPageActivity" />
-        <activity android:name=".MonitoringGroupListActivity" />
-        <activity android:name=".SearchContactsToSupervise" />
-        <activity android:name=".SendSmsInvitationActivity" />
+        <activity android:name=".viewModels.ViewAnotherUserActivity" />
+        <activity android:name=".viewModels.InvitationConfirmationActivity" />
+        <activity android:name=".activityPage.SupervisedGroupPageActivity" />
+        <activity android:name=".viewModels.SupervisedGroupListActivity" />
+        <activity android:name=".activityPage.MonitoringGroupPageActivity" />
+        <activity android:name=".viewModels.MonitoringGroupListActivity" />
+        <activity android:name=".viewModels.SearchContactsToSupervise" />
+        <activity android:name=".viewModels.SendSmsInvitationActivity" />
         <activity android:name=".activityPage.HomePageActivity" />
-        <!-- TODO The login page should load up first until we have a welcome page-->
         <activity android:name=".activityPage.LoginPageActivity"/>
-        <activity android:name=".activityPage.MainActivity">
+        <activity android:name=".activityPage.MainPageActivity"/>
+        <!--The login page should load up first-->
+        <activity android:name=".activityPage.WelcomePageActivity">
             <intent-filter>
                 <action android:name="android.intent.action.MAIN" />
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
```

Comment:
The reason why I leave it to the welcomePage, its because we want to load up the welcome page first. After that, it would be up to the welcome page to decide which page it goes.

Reply:
Fair enough.